central/splunk/compliance_test.go

```diff
@@ -0,0 +1,188 @@
+							"CIS_Docker_v1_2_0:5_6": {
+								Evidence: []*storage.ComplianceResultValue_Evidence{{
+									State:   storage.ComplianceState_COMPLIANCE_STATE_SUCCESS,
+									Message: "Container has no ssh process running",
+								}},
+								OverallState: storage.ComplianceState_COMPLIANCE_STATE_SUCCESS,
+							},
+						},
+					},
+				},
+				NodeResults: map[string]*storage.ComplianceRunResults_EntityResults{
+					"node1": {
+						ControlResults: map[string]*storage.ComplianceResultValue{
+							"CIS_Docker_v1_2_0:1_1_2": {
+								Evidence: []*storage.ComplianceResultValue_Evidence{{
+									State:   storage.ComplianceState_COMPLIANCE_STATE_SKIP,
+									Message: "Node does not use Docker container runtime",
+								}},
+								OverallState: storage.ComplianceState_COMPLIANCE_STATE_SKIP,
+							},
+						},
+					},
+				},
+				MachineConfigResults: map[string]*storage.ComplianceRunResults_EntityResults{
+					"ocp4-cis-node-master": {
+						ControlResults: map[string]*storage.ComplianceResultValue{
+							"ocp4-cis-node:file-owner-worker-kubeconfig": {
+								Evidence: []*storage.ComplianceResultValue_Evidence{{
+									State:   storage.ComplianceState_COMPLIANCE_STATE_SUCCESS,
+									Message: "Pass for ocp4-cis-node-master-file-owner-worker-kubeconfig.",
+								}},
+								OverallState: storage.ComplianceState_COMPLIANCE_STATE_SUCCESS,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+)
+
+func TestSplunkComplianceAPI(t *testing.T) {
+	suite.Run(t, &splunkComplianceAPITestSuite{})
+}
+
+type splunkComplianceAPITestSuite struct {
+	suite.Suite
+
+	hasReadCtx  context.Context
+	hasWriteCtx context.Context
+
+	mockCtrl *gomock.Controller
+	mockDS   *mocks.MockDataStore
+}
+
+func (s *splunkComplianceAPITestSuite) SetupTest() {
+	s.hasReadCtx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Compliance)))
+	s.mockCtrl = gomock.NewController(s.T())
+	s.mockDS = mocks.NewMockDataStore(s.mockCtrl)
+}
+
+func (s *splunkComplianceAPITestSuite) TearDownTest() {
+	s.mockCtrl.Finish()
+}
+
+func (s *splunkComplianceAPITestSuite) TestComplianceAPIResults() {
+	// set up http mocks
+	req, err := http.NewRequest("GET", "/api/splunk/ta/compliance", nil)
+	require.NoError(s.T(), err)
+	responseRecorder := httptest.NewRecorder()
+
+	// set up override for getClusterIDs
+	getMockClusterIDs := func(ctx context.Context) ([]string, error) {
+		return clusterIDs, nil
+	}
+
+	// configure storage mock
+	s.mockDS.EXPECT().GetLatestRunResultsBatch(req.Context(), clusterIDs, gomock.Any(), types.RequireMessageStrings).Return(latestRunResultBatch, nil).AnyTimes()
+
+	// use internal function that accepts an additional parameter to simplify mocking/testing
+	handler := newComplianceHandler(s.mockDS, getMockClusterIDs)
+
+	handler.ServeHTTP(responseRecorder, req)
+	responseBody := responseRecorder.Body.String()
```

Comment:
[nit] This could be extracted into a function eg. createMockResponseBody

Reply:
True, I could move this into its own function, but as we're only using this once for one test case, I don't know whether this would increase readability or not. I'd tend to lean towards leaving it like this.